File: training/src/classesandobjects/java/Customer2.java
package classesandobjects.java;

public class Customer2 {
    private String id;
    private final String name;
    private final String address;

    public Customer2(String name, String address) {
        this.name = name;
        this.address = address;
    }

    public void setId(String id) {
        this.id = id;
    }

    public static void main(String[] args) {
        Customer2 bob = new Customer2("Bob", "Alvear 1520");
        bob.setId("0001");
    }
}

File: training/src/classesandobjects/scala/Customer2.scala
package classesandobjects.scala

class Customer2(val name: String, val address: String) {
  var id: String = ""
}

object Customer2 {
  def main(args: Array[String]): Unit = {
    val bob = new Customer2("bob", "Alvear 1520")
    bob.id = "0001"
    bob.id_$eq("0001")
  }
}